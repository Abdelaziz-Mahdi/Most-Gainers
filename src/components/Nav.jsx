import { NavLink, Outlet } from 'react-router-dom';
import settingIcon from '../assets/settings.png';
import '../styles/Nav.css';

export default function Nav() {
  return (
    <header className="">
      <nav className="nav-bar">
        <ul className="nav-list">
          <li className="">
            <NavLink
              to="/"
              className="nav-title"
            >
              &lt;
            </NavLink>
          </li>
          <li>
            <p className="nav-title">Stock Market Most Gainers</p>
          </li>
          <li className="">
            <img src={settingIcon} alt="back-icon" width="24" height="24" />
          </li>
        </ul>
      </nav>
      <Outlet />
    </header>
  );
}
