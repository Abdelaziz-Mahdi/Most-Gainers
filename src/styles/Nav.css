.nav-bar {
  background-color: rgb(204, 78, 137);
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  height: 3.5rem;
  position: sticky;
  top: 0;
  width: 100%;
  padding: 0.5rem 1rem;
}

li.nav-li {
  display: flex;
  flex-direction: row;
}

ul.nav-list,
p.nav-title {
  margin-bottom: 0;
}

.nav-bar ul {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.nav-title {
  color: #fff;
  text-decoration: none;
}
