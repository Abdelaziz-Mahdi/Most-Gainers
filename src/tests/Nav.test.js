import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Nav from '../components/Nav';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Nav />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
