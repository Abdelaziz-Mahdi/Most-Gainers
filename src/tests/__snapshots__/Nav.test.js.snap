// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className=""
>
  <nav
    className="nav-bar"
  >
    <ul
      className="nav-list"
    >
      <li
        className=""
      >
        <a
          aria-current="page"
          className="nav-title active"
          href="/"
          onClick={[Function]}
        >
          &lt;
        </a>
      </li>
      <li>
        <h2
          className="nav-title"
        >
          Stock Market Most Gainers
        </h2>
      </li>
      <li
        className=""
      >
        <img
          alt="back-icon"
          height="24"
          src="settings.png"
          width="24"
        />
      </li>
    </ul>
  </nav>
</header>
`;
